source/agora/consensus/protocol/Nominator.d

```diff
@@ -534,6 +542,7 @@ extern(D):
     {
         const cur_time = this.clock.networkTime();
         const exp_time = this.getExpectedBlockTime(Height(slot_idx));
+        ValidationLevel temp_level;
```

Comment:
We don't need a state variable for this, we can use early returns. I'll add more comments below.